Models/TreeModel.cpp

```diff
@@ -189,24 +190,17 @@ bool TreeModel::dropMimeData(const QMimeData *mimeData, Qt::DropAction action, i
 
   for (auto index : nodes) {
     auto node = IndexToNode(index);
-    auto oldParent = index.parent();
 
     auto msg = node->GetMessage();
     Message* m_copy = msg.New();
     m_copy->CopyFrom(msg);
     messages.push_back(m_copy);
-
-    if (action == Qt::MoveAction) {
-      // offset the row to insert at by the number of
-      // rows already removed from the same parent
-      if (parent == oldParent && index.row() < row)
-        --row;
-    }
   }
 
   if (action == Qt::MoveAction)
     BatchRemove(nodes);
```

Comment:
@JoshDreamland This change crashes because you blow away the tree here instead of allowing the persistent model indexes to be repaired.